app/controllers/api/v2/courses_controller.rb

```diff
@@ -19,16 +19,6 @@ def show
         render jsonapi: @course, include: params[:include]
       end
 
-      def sync_with_search_and_compare
-        response = ManageCoursesAPIService::Request.sync_course_with_search_and_compare(
-          @current_user.email,
-          @provider.provider_code,
-          @course.course_code
-        )
-
-        head response ? :ok : :internal_server_error
-      end
-
```

Comment:
How about instead of killing it, and as a result whatever is calling it, we just call `sync_with_search_and_compare` from `publish` below?